x/profiles/keeper/keeper.go

```diff
@@ -55,6 +68,12 @@ func (k Keeper) IsUserBlocked(ctx sdk.Context, blocker, blocked string) bool {
 // It assumes that the given profile has already been validated.
 // It returns an error if a profile with the same DTag from a different creator already exists
 func (k Keeper) StoreProfile(ctx sdk.Context, profile *types.Profile) error {
+
+	if _, found := k.GetAccountByChainLink(ctx, types.DesmosChainName, profile.GetAddress().String()); found {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest,
+			"owner address has already been linked to another profile")
+	}
```

Comment:
We can remove this as we do not care anymore if the destination link already has a profile